contracts/ReputationMiningCycle.sol

```diff
@@ -274,7 +278,7 @@ contract ReputationMiningCycle is ReputationMiningCycleStorage, PatriciaTreeProo
       require(disputeRounds[round][opponentIdx].challengeStepCompleted >= disputeRounds[round][idx].challengeStepCompleted, "colony-reputation-mining-less-challenge-rounds-completed");
 
       // Require that it has failed a challenge (i.e. failed to respond in time)
-      require(now - disputeRounds[round][idx].lastResponseTimestamp >= 600, "colony-reputation-mining-not-timed-out"); // Timeout is ten minutes here.
+      require(disputeRounds[round][idx].lastResponseTimestamp < now && sub(now, disputeRounds[round][idx].lastResponseTimestamp) >= 600, "colony-reputation-mining-not-timed-out"); // Timeout is ten minutes here.
```

Comment:
`lastResponseTimestamp` is set to a value in the future when each submission is made - specifically, when the window is going to close. By doing that, and having these checks, we make sure we're not accidentally paying anyone out before we know how many submissions there have been (and therefore knowing how much the reward should be for each). I will admit that I have not properly considered whether this particular one can be hit - if that's your specifically the question here, let me know and I'll look properly.

Reply:
Hmm, ok that makes sense, I didn't realize it was a future cutoff, I thought it was the time of a prior submission. But re-working the arithmetic so that `now` is always on the same side of the inequality would be clarifying, I think. So instead of:

```
require(
  disputeRounds[round][idx].lastResponseTimestamp < now &&
  sub(now, disputeRounds[round][idx].lastResponseTimestamp) >= 600,
  "colony-reputation-mining-not-timed-out"
); // Timeout is ten minutes here.
```

we have the equivalent (since `now - ts > x <=> now > ts + x`):

```
require(
  disputeRounds[round][idx].lastResponseTimestamp < now &&
  add(disputeRounds[round][idx].lastResponseTimestamp, 600) <= now,
  "colony-reputation-mining-not-timed-out"
); // Timeout is ten minutes here.
```

So we're checking that the submission window has closed (`lastResponseTimestamp < now`) AND that the 10-minute timeout for response has elapsed (`add(lastResponseTimestamp, 600) <= now`). Looking at it this way, it does seem like the first check is redundant -- if the second passes, the first one will also.

Reply:
To be clear, once a dispute has started, it does represent the last time that submission was defended as part of that process. I agree with the rest of this though :smile: